lib/logstash/inputs/exec.rb

```diff
@@ -33,44 +33,50 @@ def register
   end # def register
 
   def run(queue)
-
     while !stop?
+      inner_run(queue)
+    end # loop
+  end # def run
+
+  def inner_run(queue)
       start = Time.now
-      execute(@command)
+      execute(@command, queue)
       duration = Time.now - start
 
       @logger.info? && @logger.info("Command completed", :command => @command, :duration => duration)
 
       wait_until_end_of_interval(duration)
-    end # loop
-  end # def run
+  end
 
   def stop
-    if @io
-      @io.close
-      @io = nil
-    end
+    close_and_nilify_io
   end
 
   private
 
+  def close_and_nilify_io
+    return if @io.nil? || @io.closed?
+    @io.close
+  end
```

Comment:
Not sure about the name here since we are actually only closing the io and don't really **nilify** things?

On another note, maybe this logic could stay in the `#stop` method and then https://github.com/logstash-plugins/logstash-input-exec/pull/7/files#diff-d0e603d746d065da28958716df229acfR94 could call `#stop`, what do you think? Because the way `#stop` and `#close_and_nilify_io` are structured they could be replaced with an alias_method, since they are not different other than the name and the visibility.